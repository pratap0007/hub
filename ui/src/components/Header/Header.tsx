import React from 'react';
import '@patternfly/react-core/dist/styles/base.css';
import {
  PageHeader,
  Brand,
  PageHeaderTools,
  Text,
  TextVariants,
  GridItem,
  Grid
} from '@patternfly/react-core';
import logo from '../../assets/logo/logo.png';
import SearchBar from '../SearchBar/SearchBar';

const Header: React.FC = () => {
  const logoProps = {
    href: '/',
    target: ''
  };

  const headerTools = (
    <PageHeaderTools>
      <Grid>
        <GridItem span={11}>
          <SearchBar />
        </GridItem>
      </Grid>
      <Text component={TextVariants.h3}>Login</Text>
    </PageHeaderTools>
  );

  return (
    <PageHeader
      logo={<Brand src={logo} alt="Tekton Hub Logo" />}
      headerTools={headerTools}
      logoProps={logoProps}
    />
  );
};

export default Header;
