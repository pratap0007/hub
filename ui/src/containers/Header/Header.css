.hub-header-login {
  color: white;
  fontsize: 1em;
}
