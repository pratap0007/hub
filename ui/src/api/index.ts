import axios from 'axios';
import { API_URL } from '../config/constants';
import { ICategory } from '../store/category';
import { IResource, IVersion } from '../store/resource';

interface Token {
  token: string;
  refreshInterval: string;
  expiresAt: number;
}

interface TokenData {
  access: Token;
  refresh: Token;
}

export interface AuthResponse {
  data: TokenData;
}

export interface Api {
  categories(): Promise<ICategory>;
  resources(): Promise<IResource>;
  resourceVersion(resourceId: number): Promise<IVersion>;
  versionUpdate(versionId: number): Promise<IVersion>;
  authentication(authCode: string): Promise<AuthResponse>;
}

export class Hub implements Api {
  async categories() {
    try {
      return axios.get(`${API_URL}/categories`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resources() {
    try {
      return axios.get(`${API_URL}/resources`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async authentication(authCode: string) {
    try {
      return axios.post(`${API_URL}/auth/login?code=${authCode}`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resourceVersion(resourceId: number) {
    try {
      return axios
        .get(`${API_URL}/resource/${resourceId}/versions`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async versionUpdate(versionId: number) {
    try {
      return axios
        .get(`${API_URL}/resource/version/${versionId}`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }
}
