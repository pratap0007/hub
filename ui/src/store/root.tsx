import React, { ReactChild, ReactChildren } from 'react';
import { types, getEnv, Instance } from 'mobx-state-tree';
import { persist } from 'mst-persist';
import { CategoryStore } from './category';
import { ResourceStore } from './resource';
import { Hub, Api } from '../api';
import { AuthStore } from './auth';

export const Root = types.model('Root', {}).views((self) => ({
  get api(): Api {
    return getEnv(self).api;
  },
  get categories() {
    return getEnv(self).categories;
  },
  get resources() {
    return getEnv(self).resources;
  },
  get user() {
    return getEnv(self).user;
  }
}));

type IRoot = Instance<typeof Root>;

const initRootStore = (api: Api) => {
  const categories = CategoryStore.create({}, { api });
  const resources = ResourceStore.create({}, { api, categories });
  const user = AuthStore.create({ accessTokenInfo: {}, refreshTokenInfo: {} }, { api });

  // This peristently stores user auth details
  persist('authStore', user, {
    storage: localStorage,

    // This adds fields from the store for which data needs to be stored persistently
    whitelist: ['accessTokenInfo', 'refreshTokenInfo', 'isAuthenticated', 'isLoading']
  });

  return Root.create({}, { api, categories, resources, user });
};

interface Props {
  children: ReactChild | ReactChildren;
}

let RootContext: React.Context<IRoot>;
export const useMst = () => React.useContext(RootContext);

export const createProviderAndStore = (api?: Api) => {
  const root = initRootStore(api || new Hub());
  RootContext = React.createContext(root);

  const Provider = ({ children }: Props) => (
    <RootContext.Provider value={root}> {children} </RootContext.Provider>
  );
  return { Provider, root };
};

export const createProvider = (api?: Api) => {
  const { Provider } = createProviderAndStore(api);
  return Provider;
};
