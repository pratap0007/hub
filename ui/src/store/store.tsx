import React from "react";
import {CategoryStore} from "./category";
import {Hub} from "../api";

export const StoreContext = React.createContext(CategoryStore.create());

const api = new Hub()
export const StoreProvider = ({children}: any) => {
    const store = CategoryStore.create({}, {api});

    setInterval(function () {
        // console.log("---==", getSnapshot(store));
    });
    return (
        <StoreContext.Provider value={store}>{children}</StoreContext.Provider>
    );
};