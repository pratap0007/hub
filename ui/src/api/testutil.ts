import { Api } from './';
import * as fs from 'fs';
import { ICategory } from '../store/category';

export class FakeHub implements Api {
  dataDir: string;

  constructor(dataDir: string) {
    this.dataDir = dataDir;
  }

  async categories() {
    const data = `${this.dataDir}/categories.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<ICategory>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }
}
