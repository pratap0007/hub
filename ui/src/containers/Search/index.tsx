import { TextInput } from '@patternfly/react-core';
import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import { useMst } from '../../store/root';
import './Search.css';

const Search: React.FC = () => {
  const { resources } = useMst();

  const [value, setValue] = useState('');

  const onSearchChange = (text: string) => {
    setValue(text);
    resources.setSearch(text.trim());
  };

  const history = useHistory();
  const onSearchKeyPress = (e: React.KeyboardEvent<HTMLElement>) => {
    if (e.key === 'Enter') {
      e.preventDefault();
      history.push('/');
    }
    return;
  };

  return (
    <TextInput
      value={value}
      type="search"
      onChange={onSearchChange}
      onKeyPress={onSearchKeyPress}
      aria-label="text input example"
      placeholder="Search for resources..."
      spellCheck="false"
      className="hub-search"
    />
  );
};

export default Search;
