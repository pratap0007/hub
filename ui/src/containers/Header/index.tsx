import React from 'react';
import { Link } from 'react-router-dom';
import { observer } from 'mobx-react';
import { useHistory } from 'react-router-dom';
import {
  PageHeader,
  Brand,
  PageHeaderTools,
  Text,
  TextVariants,
  GridItem,
  Grid
} from '@patternfly/react-core';
import logo from '../../assets/logo/logo.png';
import Search from '../../containers/Search';
import UserProfile from '../UserProfile';
import { useMst } from '../../store/root';
import './Header.css';

const Header: React.FC = observer(() => {
  const { user } = useMst();
  const history = useHistory();

  const headerTools = (
    <PageHeaderTools>
      <Grid>
        <GridItem span={11}>
          <Search />
        </GridItem>
      </Grid>
      {user.isAuthenticated ? (
        <UserProfile />
      ) : (
        <Text component={TextVariants.h3}>
          <Link to="/login" style={{ textDecoration: 'none' }}>
            <span className="hub-header-login">Login</span>
          </Link>
        </Text>
      )}
    </PageHeaderTools>
  );

  return (
    <React.Fragment>
      <PageHeader
        logo={<Brand src={logo} alt="Tekton Hub Logo" onClick={() => history.push('/')} />}
        headerTools={headerTools}
      />
    </React.Fragment>
  );
});
export default Header;
