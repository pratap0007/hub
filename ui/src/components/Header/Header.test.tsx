import React from 'react';
import { mount } from 'enzyme';
import { FakeHub } from '../../api/testutil';
import { createProviderAndStore } from '../../store/root';
import Header from '.';
import Search from '../../containers/Search';

const TESTDATA_DIR = `src/store/testdata`;
const api = new FakeHub(TESTDATA_DIR);
const { Provider } = createProviderAndStore(api);

it('should render the header component and finds Search component', () => {
  const component = mount(
    <Provider>
      <Header />
    </Provider>
  );

  expect(component.find(Search).length).toBe(1);
  expect(component.debug()).toMatchSnapshot();
});
