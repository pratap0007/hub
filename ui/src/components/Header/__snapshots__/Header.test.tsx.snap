// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the header component and finds Search component 1`] = `
"<Provider>
   
  <Header>
    <PageHeader logo={{...}} headerTools={{...}}>
      <header role={[undefined]} className=\\"pf-c-page__header\\">
        <div className=\\"pf-c-page__header-brand\\">
          <a className=\\"pf-c-page__header-brand-link\\">
            <Brand src=\\"logo.png\\" alt=\\"Tekton Hub Logo\\" onClick={[Function: onClick]}>
              <img onClick={[Function: onClick]} className=\\"pf-c-brand\\" src=\\"logo.png\\" alt=\\"Tekton Hub Logo\\" />
            </Brand>
          </a>
        </div>
        <PageHeaderTools>
          <div className=\\"pf-c-page__header-tools\\">
            <Grid>
              <div className=\\"pf-l-grid\\">
                <GridItem span={11}>
                  <div className=\\"pf-l-grid__item pf-m-11-col\\">
                    <Search>
                      <ForwardRef value=\\"\\" type=\\"search\\" onChange={[Function: onSearchChange]} onKeyPress={[Function: onSearchKeyPress]} aria-label=\\"text input example\\" placeholder=\\"Search for resources...\\" spellCheck=\\"false\\" className=\\"hub-search\\">
                        <TextInputBase value=\\"\\" type=\\"search\\" onChange={[Function: onSearchChange]} onKeyPress={[Function: onSearchKeyPress]} aria-label=\\"text input example\\" placeholder=\\"Search for resources...\\" spellCheck=\\"false\\" className=\\"hub-search\\" innerRef={{...}} isRequired={false} validated=\\"default\\" isDisabled={false} isReadOnly={false} isLeftTruncated={false}>
                          <input onKeyPress={[Function: onSearchKeyPress]} aria-label=\\"text input example\\" placeholder=\\"Search for resources...\\" spellCheck=\\"false\\" onFocus={[Function (anonymous)]} onBlur={[Function (anonymous)]} className=\\"pf-c-form-control hub-search\\" onChange={[Function (anonymous)]} type=\\"search\\" value=\\"\\" aria-invalid={false} required={false} disabled={false} readOnly={false} />
                        </TextInputBase>
                      </ForwardRef>
                    </Search>
                  </div>
                </GridItem>
              </div>
            </Grid>
            <Text component=\\"h3\\">
              <h3 data-pf-content={true} className=\\"\\">
                Login
              </h3>
            </Text>
          </div>
        </PageHeaderTools>
      </header>
    </PageHeader>
  </Header>
   
</Provider>"
`;
